AIPscan/Data/report_data.py

```diff
@@ -631,3 +634,110 @@ def storage_locations(storage_service_id, start_date, end_date):
     report[fields.FIELD_LOCATIONS] = _sort_storage_locations(unsorted_results)
 
     return report
+
+
+def _get_months_covered_by_date_range(storage_service_id, start_date, end_date):
+    """Return months covered by date range as list of ISO values like YYYY-MM.
+
+    First, replace datetime.min and datetime.max, if used, with more sensible
+    values so that we do not end up returning every month since the beginning
+    of time.
+
+    We use the first day of each month in the calculation to prevent rrule from
+    returning odd results.
+
+    :param storage_service_id: Storage Service ID (int)
+    :param start_date: Inclusive AIP creation start date
+        (datetime.datetime object)
+    :param end_date: Inclusive AIP creation end date
+        (datetime.datetime object)
+
+    :returns: List of strings like "YYYY-MM"
+    """
+    if start_date == datetime.min:
+        storage_service = StorageService.query.get(storage_service_id)
+        start_date = storage_service.earliest_aip_created
+    if end_date >= datetime.now():
+        end_date = datetime.now()
+
+    return [
+        dt.isoformat()[:7]
+        for dt in rrule(
+            MONTHLY, dtstart=start_date.replace(day=1), until=end_date.replace(day=1)
+        )
+    ]
+
+
+def storage_locations_by_month(
+    storage_service_id, start_date, end_date, cumulative=False
+):
+    """Return details of AIP store locations in Storage Service by month.
+
+    :param storage_service_id: Storage Service ID (int)
+    :param start_date: Inclusive AIP creation start date
+        (datetime.datetime object)
+    :param end_date: Inclusive AIP creation end date
+        (datetime.datetime object)
+    :param cumulative: Flag indicating whether to calculate cumulatively, where
+        each month adds to previous totals (bool)
+
+    :returns: "report" dict containing following fields:
+        report["StorageName"]: Name of Storage Service queried
+        report["Locations"]: List of result locations ordered desc by size
+    """
+    report = {}
+    report[fields.FIELD_STORAGE_NAME] = get_storage_service_name(storage_service_id)
+
+    locations = _get_storage_locations(storage_service_id)
+    months = _get_months_covered_by_date_range(storage_service_id, start_date, end_date)
+
+    results = {}
+
+    for month in months:
+        monthly_locations_data = []
+
+        for location in locations:
+            loc_info = {}
+
+            # Determine local start and end date values for the subquery.
+            # For start date, use first day of month or start_date, if later
+            # and in the same month. If cumulative, always use start date.
+            # For end date, sue last day of month or end_date, if prior and in
+            # the same month.
+            local_start = datetime.strptime(month, "%Y-%m").replace(day=1)
+
+            if start_date.isoformat()[:7] == month and start_date > local_start:
+                local_start = start_date
```

Comment:
Hm, I hear what you mean. Maybe I'll try setting the chart resolution per-day and see how manageable that is for large reporting periods (the client also expressed some interest being able to do this, though we both wondered if it would end up being too granular). Will try some things and then run it past you for a second opinion when I have something that seems like it's working well.

Reply:
@djjuhasz commit https://github.com/artefactual-labs/AIPscan/pull/177/commits/2dbcdb7697f188b7324a8087b30999712cf6885c changes the reporting interval to daily - want to give it a spin and let me know what you think of the user experience? It does make the code a fair bit simpler in a good way, and I think will address your concern about user confusion.

I haven't updated the tests yet - figured I'd see what you think of the change first. Thanks!